### ISSUE TITLE
No tests found to run when TestFixture is a nested class

### ISSUE THREAD

**Issue by rprouse** [nunit/nunit3-vs-adapter#218]
@pitlyck commented on [Thu Aug 11 2016](https://github.com/nunit/nunit-vs-adapter/issues/126)

Hi,

I am unable to run tests by selecting "Run Tests" from the context menu in VS2015 when the test fixture is a nested class.

Right-click in the window with code below, then select "Run Tests". In output window I can see output from tests: **No tests found to run.** 

Example:

```
public class NestedClasses
    {
        [TestFixture]
        public class NestedClass1
        {
            [Test]
            public void Test1()
            {
                Assert.Pass();
            }
        }
    }
```

I can see the test in list of tests in Test Explorer window and I am able to run it from there, but this is not what I want to do when I need to run/debug specific test/fixture that I am currently working on.

I am sure it worked before switching from NUnit 2.6.4 to NUnit 3.2.0.

I tried similar scenario with Microsoft VisualStudio UnitTesting framework and the test was run properly, so I expect this is nunit-vs-adapter issue.

VS 2015 Update 3 + NUnit3 Test Adapter v3.4.1 + NUnit 3.2.0

---

@rprouse commented on [Thu Aug 11 2016](https://github.com/nunit/nunit-vs-adapter/issues/126#issuecomment-239179975)

Since this is an issue with the NUnit 3 adapter, I am moving this issue to that repository.

**Comment by CharliePoole** [comment]
I'm guessing that the bottom panel of the test window shows "No source found" for these tests. Is that correct?

**Comment by pitlyck** [comment]
Yes, when I select the test in Test Explorer window the following message is displayed at the bottom panel: "Source: no source available"

**Comment by CharliePoole** [comment]
I think this has to do with the difference between how nested classes are represented using Reflection versus Mono.Cecil. 

Reflection uses a '+' between the outer class and the inner one. Cecil uses the ECMA-235 representation, which specifies a '/' to separate them.

We are getting the class in reflection format from NUnit and then using it to ask the engine and locate the source code. The engine uses Cecil.

We need to change the format somewhere but it will take some thought to decide whether this should be done in the framework, the engine or the adapter!

**Comment by CharliePoole** [comment]
When working on this, we should probably look at other differences caused by Cecil's use of ECMA-235 and deal with them as well. I believe that the repreentation of generics may differ.

**Comment by jblazek** [comment]
We also have run into this issue after upgrading from NUnit2 to 3.  Another symptom of the issue is double-clicking on a test name in the Test Explorer does not navigate to the source code.

Regarding where to make this fix (framework, engine, or adapter), it looks like there is already logic in the adapter's [NavigationDataProvider](https://github.com/nunit/nunit3-vs-adapter/blob/master/src/NUnitTestAdapter/NavigationDataProvider.cs#L153), but it doesn't get called everywhere types are looked up from the type cache. 

Specifically, this standardize name call is used when matching methods on [line 113](https://github.com/nunit/nunit3-vs-adapter/blob/master/src/NUnitTestAdapter/NavigationDataProvider.cs#L113), but not when matching types initially on [line 51](https://github.com/nunit/nunit3-vs-adapter/blob/master/src/NUnitTestAdapter/NavigationDataProvider.cs#L51).

I could submit a PR if you are accepting them.

**Comment by CharliePoole** [comment]
Yes, we love PRs!

**Comment by rprouse** [comment]
Fixed by #285
